### PR TITLE
Testing strategy for components

### PR DESCRIPTION
**Pull request by lloydwaddell**
```Context```
I had an idea of creating generic testing data that could be reused for components, but thinking this is more confusing / complex than it should be. In addition the AddMock util was only good for learning paths use case. Defining mocks inside the tests with the data seems to be a better approach, so I am updated the testTemplate and started to cleanup the tests.

Leaving the PR where it is for now to discuss if this is the approach we should follow.

### CODE REVIEW

File: test/testTemplate.js [hunk 1]
```diff
@@ -58,11 +61,11 @@ describe('component-name', () => {
 		});
 
 		afterEach(() => {
-			mockLink.resetHistory();
+			fetchMock.resetHistory();
 		});
 
 		it('should ...', async() => {
-			await _createComponent('path');
+			await _createComponent(apiHref);
```
Comment (AKobets) [inline test/testTemplate.js:68]: is this test needed? See similar test at line 46

File: components/activity/code/test/d2l-activity-code-editor.test.js [hunk 2]
```diff
@@ -50,29 +80,18 @@ describe('d2l-activity-code-editor', async() => {
...
-		});
-
-		it('should initialize using defined path and expected values', async() => {
-			const element = await _createCodeEditor('/learning-path/new');
-
-			assert.equal(element.code, learningPathNew.properties.code);
-		});
-
-		describe('path:/learning-path/existing', () => {
+		describe('Code', () => {
```
Comment (AKobets) [inline components/activity/code/test/d2l-activity-code-editor.test.js:83]: would it be better `describe('d2l-activity-code-editor',...`

File: components/activity/code/test/d2l-activity-code-editor.test.js [hunk 3]
```diff
@@ -100,18 +119,20 @@ describe('d2l-activity-code-editor', async() => {
...
 		});
-		describe('path:/learning-path/missing-action', () => {
-			let element;
-			beforeEach(async() => {
-				clearStore();
-				element = await _createCodeEditor('learning-path/missing-action');
-				assert.equal(element.code, learningPathMissingAction.properties.code, 'code should match response');
-			});
+
+		describe('Update code action missing', () => {
```
Comment (AKobets) [inline components/activity/code/test/d2l-activity-code-editor.test.js:123]: Does `code not updated when action missing` test needs its own describe section?

File: components/activity/code/test/d2l-activity-code-editor.test.js [hunk 4]
```diff
@@ -100,18 +119,20 @@ describe('d2l-activity-code-editor', async() => {
...
+
 			it('code not updated when action missing', async() => {
+				fetchMock.mock(orgHref, JSON.stringify(orgNoAction));
+
+				clearStore();
```
Comment (AKobets) [inline components/activity/code/test/d2l-activity-code-editor.test.js:128]: It looks like clearStore is already called at line 66

File: components/activity/code/test/d2l-activity-code-editor.test.js (hunk 3 above)
Comment (lloydwaddell) [reply components/activity/code/test/d2l-activity-code-editor.test.js:123]: I guess not lol

**Review by AKobets — APPROVED** [review]
Looks good.

**Pull request merged**